tools/clang/test/HLSLFileCheck/shader_targets/library/vec_array_param.hlsl

```diff
@@ -0,0 +1,11 @@
+// RUN: %dxc -T lib_6_6 %s | FileCheck %s
+
+// Make sure vector array param works.
+// CHECK:call float @"\01?foo@@YAMY02V?$vector@M$02@@@Z"([3 x <3 x float>]* %{{.+}})
```

Comment:
This test doesn’t actually verify the correct transformation occurs, it just verifies we don’t hit the assert.

We should test things like this using IR tests either through `dxopt` or `opt`.

Reply:
Fixed.